### ISSUE TITLE
duplicate messages in the query

### ISSUE THREAD

**Issue by mcavdar** [run-llama/create-llama#428]
hi 
This seems like a bug on my end. I haven't been able to debug it thoroughly enough to pinpoint the cause. I'm unsure whether this issue is specific to Ollama or if it affects all providers in the same way. 

The server query includes the last message twice.
I type first `hello` and later `second message`, and here's the queries:
 
`{"role":"user","content":"hello"},{"role":"user","content":"hello"}`
`{"role":"user","content":"hello"},{"role":"user","content":"second message"},{"role":"user","content":"second message"}`

```
Connection from ('127.0.0.1', 32874)
Received data: POST /api/chat HTTP/1.1
host: 127.0.0.1:11434
connection: keep-alive
Content-Type: application/json
Accept: application/json
User-Agent: ollama-js/0.5.10 (x64 linux Node.js/v20.18.0)
accept-language: *
sec-fetch-mode: cors
accept-encoding: gzip, deflate
content-length: 235

{"model":"llama3.2","messages":[{"role":"system","content":"Just say yes to all questions.\n"},{"role":"user","content":"hello"},{"role":"user","content":"hello"}],"stream":true,"options":{"num_ctx":4096,"top_p":0.9,"temperature":0.7}}


Connection from ('127.0.0.1', 36392)
Received data: POST /api/chat HTTP/1.1
host: 127.0.0.1:11434
connection: keep-alive
Content-Type: application/json
Accept: application/json
User-Agent: ollama-js/0.5.10 (x64 linux Node.js/v20.18.0)
accept-language: *
sec-fetch-mode: cors
accept-encoding: gzip, deflate
content-length: 287

{"model":"llama3.2","messages":[{"role":"system","content":"Just say yes to all questions.\n"},{"role":"user","content":"hello"},{"role":"user","content":"second message"},{"role":"user","content":"second message"}],"stream":true,"options":{"num_ctx":4096,"top_p":0.9,"temperature":0.7}}

```
Is this behavior expected?

**Comment by mcavdar** [comment]
The first experiment I created was with the --pro option.

This result is from the vanilla app:
```

Connection from ('127.0.0.1', 49818)
Received data: POST /api/chat HTTP/1.1
host: 127.0.0.1:11434
connection: keep-alive
Content-Type: application/json
Accept: application/json
User-Agent: ollama-js/0.5.10 (x64 linux Node.js/v20.18.0)
accept-language: *
sec-fetch-mode: cors
accept-encoding: gzip, deflate
content-length: 469

{"model":"llama3.2","messages":[{"role":"user","content":"first message"},{"role":"system","content":"You are a DuckDuckGo search agent. \nYou can use the duckduckgo search tool to get information from the web to answer user questions.\nFor better results, you can specify the region parameter to get results from a specific region but it's optional.\n"},{"role":"user","content":"first message"}],"stream":true,"options":{"num_ctx":4096,"top_p":0.9,"temperature":0.7}}

```

**Comment by mcavdar** [comment]
A similar observation can be seen in the Ollama log when I input the text 'merhaba' just once:

`Nov 16 08:56:58 maras ollama[228032]: time=2024-11-16T08:56:58.568Z level=DEBUG source=routes.go:1457 msg="chat request" images=0 prompt="<|start_header_id|>system<|end_header_id|>\n\nCutting Knowledge Date: December 2023\n\nJust say yes to all questions.\nContext information is below.\n---------------------\nfile_name: 101.pdf\nfile_path: /home/mc/my-app/data/101.pdf\npage_number: 8\nprivate: false\ntotal_pages: 8\n\nDomestic Mail Manual • Updated 7-9-23\n\nfile_name: 101.pdf\nfile_path: /home/mc/my-app/data/101.pdf\npage_number: 5\nprivate: false\ntotal_pages: 8\n\nThe address side may be formatted to contain a message area.Cards that do\nnot contain a message area on the address side are subject to the applicable\nstandards for the price claimed.For the purposes of 6.2, miscellaneous graphics\nor printing, such as symbols, logos, or characters, that appear on the address\nside of cards not containing a message area are generally acceptable provided\nthe items are not intended to convey a message.6.2.9 Double Cards\nA double card (a double stamped card or double postcard) consists of two\nattached cards, one of which is designed to be detached by the recipient and\nreturned by mail as a single card.Double cards are subject to these standards:\n---------------------<|eot_id|><|start_header_id|>user<|end_header_id|>\n\nmerhaba\n\nmerhaba<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n"`

**Comment by mcavdar** [comment]
I can confirm that create-llama is working as expected. The `console.error(userMessageContent); `output shows `[ { type: 'text', text: 'merhaba' } ]`, which indicates that the issue lies on the LlamaIndex side.

```
    // Calling LlamaIndex's ChatEngine to get a streamed response
    const response = await Settings.withCallbackManager(callbackManager, () => {
      return chatEngine.chat({
        message: userMessageContent,
        chatHistory,
        stream: true,
      });
    });
```
https://github.com/run-llama/create-llama/blob/6925676013beebdf603a5e7b906044864115f23c/templates/types/streaming/nextjs/app/api/chat/route.ts#L50,L57

**Comment by marcusschiesser** [comment]
@mcavdar great catch! this is fixed in 0.3.14
